FF/FF.C

```diff
@@ -69,11 +68,31 @@ std::string preciceAdapter::FF::FluidFluid::determineSolverType()
     // NOTE: When coupling a different variable, you may want to
     // add more cases here. Or you may provide the solverType in the config.
 
-    std::string solverType;
-    
-    // TODO: Implement properly
-    solverType = "incompressible";
-    
+    std::string solverType = "unknown";
+
+    dimensionSet pressureDimensionsCompressible(1, -1, -2, 0, 0, 0, 0);
+    dimensionSet pressureDimensionsIncompressible(0, 2, -2, 0, 0, 0, 0);
+
+    if (mesh_.foundObject<volScalarField>("p"))
+    {
+      volScalarField p_ = mesh_.lookupObject<volScalarField>("p");
+
+      if (p_.dimensions() == pressureDimensionsCompressible)
+        solverType = "compressible";
+      else if (p_.dimensions() == pressureDimensionsIncompressible)
+        solverType = "incompressible";
+    }
```

Comment:
I am unfamiliar with the units for compressible and incompressible pressure, but so far I have been using an incompressible solver, and my pressure units are kgm/s^2 m^2 (1, -1, -2, 0, 0, 0, 0), isnt this the other way?

Reply:
In an incompressible solver, OpenFOAM [uses the kinematic pressure](https://www.openfoam.com/documentation/guides/latest/doc/guide-applications-solvers-variable-transform-kinematic-pressure.html), which is in `m^2 / s^2` and in named `p`. Compressible solvers use the static pressure, which is in  `Pa` or `kg / (m s^2) and is also named `p`.

You are probably talking about interFoam, which [splits the pressure](https://www.openfoam.com/documentation/guides/latest/doc/guide-applications-solvers-multiphase-interFoam.html) into `p` and `p_rgh`, both in `Pa`. We should indeed consider this as a special case when we add support for interFoam, good point!

Reply:
In any case, this seems to correctly distinguish between `pimpleFoam` (incompressible) and `sonicLiquidFoam` (compressible).

Reply:
cool, then in case for interFoam, lets lets a `TODO` there

Reply:
I quickly tried with interFoam, it classifies it as an incompressible solver. We currently don't use the solverType anyway, but I added a TODO. We will probably need it in the future.